examples/benchmarks/tablebench/src/test/table.bench.ts

```diff
@@ -0,0 +1,148 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { NodeFromSchema, SharedTree } from "@fluidframework/tree";
+import { SharedMatrix } from "@fluidframework/matrix";
+import { benchmark, BenchmarkType, isInPerformanceTestingMode } from "@fluid-tools/benchmark";
+import { IChannel } from "@fluidframework/datastore-definitions";
+import { generateTable, Table } from "..";
+import { create, measureAttachmentSummary, measureEncodedLength } from "./utils";
+
+const numRows = isInPerformanceTestingMode ? 10000 : 100;
+
+describe("Table", () => {
+	const data = generateTable(numRows);
+
+	describe(`compute over ${numRows} rows`, () => {
+		let table: NodeFromSchema<typeof Table>;
+		let matrix: SharedMatrix;
+
+		let channel: IChannel;
+		let processAllMessages: () => void;
+
+		const columnNames = Object.keys(data[0]);
+		const unitsSoldColumn = columnNames.indexOf("Units Sold");
+		const unitPriceColumn = columnNames.indexOf("Unit Price");
+		const unitCostColumn = columnNames.indexOf("Unit Cost");
+		const totalRevenueColumn = columnNames.indexOf("Total Revenue");
+		const totalCostColumn = columnNames.indexOf("Total Cost");
+		const totalProfitColumn = columnNames.indexOf("Total Profit");
+
+		benchmark({
+			type: BenchmarkType.Measurement,
+			title: `SharedMatrix`,
+			before: () => {
+				({ channel, processAllMessages } = create(SharedMatrix.getFactory()));
+				matrix = channel as SharedMatrix;
+				matrix.insertCols(0, columnNames.length);
+				matrix.insertRows(0, data.length);
+
+				for (let r = 0; r < data.length; r++) {
+					for (const [c, key] of columnNames.entries()) {
+						matrix.setCell(r, c, (data as any)[r][key]);
+					}
+				}
+				processAllMessages();
+			},
+			benchmarkFn: () => {
+				for (let r = 0; r < matrix.rowCount; r++) {
+					const unitsSold = matrix.getCell(r, unitsSoldColumn) as number;
+					const unitPrice = matrix.getCell(r, unitPriceColumn) as number;
+					const unitCost = matrix.getCell(r, unitCostColumn) as number;
+
+					const totalRevenue = unitsSold * unitPrice;
+					const totalCost = unitsSold * unitCost;
+					const totalProfit = totalRevenue - totalCost;
+
+					matrix.setCell(r, totalRevenueColumn, totalRevenue);
+					matrix.setCell(r, totalCostColumn, totalCost);
+					matrix.setCell(r, totalProfitColumn, totalProfit);
+				}
+				processAllMessages();
+			},
+		});
+
+		benchmark({
+			type: BenchmarkType.Measurement,
+			title: `SharedTree`,
+			before: () => {
+				({ channel, processAllMessages } = create(SharedTree.getFactory()));
+				const tree = channel as SharedTree;
+
+				const view = tree.schematize({
+					schema: Table,
+					initialTree: () => data,
+				});
+
+				table = view.root;
+
+				processAllMessages();
+			},
+			benchmarkFn: () => {
+				for (const row of table) {
```

Comment:
Another future benchmark idea: put all changes in a single transaction.

Reply:
No.  We're using this code to setup the Run button on 'index.html', but the benchmark doesn't actually run until you push the button.

(Renamed to 'initApp()' instead of 'start()'.)